src/lib/detect-install.ts

```diff
@@ -0,0 +1,91 @@
+import { IgniteToolbox, IgniteDetectInstall } from '../types'
```

Comment:
This file had fairly extensive rework, but it's well-tested, so I wouldn't worry about it too much on the manual code review.